compactor/src/parquet_file_combining.rs

```diff
@@ -0,0 +1,579 @@
+
+    let parquet_chunk = ParquetChunk::new(Arc::clone(&file), Arc::new(schema), store);
+
+    trace!(
+        parquet_file_id=?file.id,
+        parquet_file_sequencer_id=?file.sequencer_id,
+        parquet_file_namespace_id=?file.namespace_id,
+        parquet_file_table_id=?file.table_id,
+        parquet_file_partition_id=?file.partition_id,
+        parquet_file_object_store_id=?file.object_store_id,
+        "built parquet chunk from metadata"
+    );
+
+    QueryableParquetChunk::new(
+        table_name,
+        file.partition_id,
+        Arc::new(parquet_chunk),
+        &[],
+        file.max_sequence_number,
+        file.min_time,
+        file.max_time,
+        sort_key,
+        partition_sort_key,
+    )
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arrow::record_batch::RecordBatch;
+    use arrow_util::assert_batches_sorted_eq;
+    use data_types::{ColumnType, PartitionKey, PartitionParam};
+    use iox_tests::util::{TestCatalog, TestParquetFileBuilder, TestTable};
+    use parquet_file::ParquetFilePath;
+    use test_helpers::assert_error;
+
+    struct TestSetup {
+        catalog: Arc<TestCatalog>,
+        table: Arc<TestTable>,
+        candidate_partition: PartitionCompactionCandidateWithInfo,
+        parquet_files: Vec<ParquetFile>,
+    }
+
+    async fn test_setup() -> TestSetup {
+        let catalog = TestCatalog::new();
+        let ns = catalog.create_namespace("ns").await;
+        let sequencer = ns.create_sequencer(1).await;
+        let table = ns.create_table("table").await;
+        table.create_column("field_int", ColumnType::I64).await;
+        table.create_column("tag1", ColumnType::Tag).await;
+        table.create_column("tag2", ColumnType::Tag).await;
+        table.create_column("tag3", ColumnType::Tag).await;
+        table.create_column("time", ColumnType::Time).await;
+        let table_schema = table.catalog_schema().await;
+
+        let partition = table
+            .with_sequencer(&sequencer)
+            .create_partition("2022-07-13")
+            .await;
+
+        // The sort key comes from the catalog and should be the union of all tags the
+        // ingester has seen
+        let sort_key = SortKey::from_columns(["tag1", "tag2", "tag3", "time"]);
+        let partition = partition.update_sort_key(sort_key.clone()).await;
+
+        let candidate_partition = PartitionCompactionCandidateWithInfo {
+            table: Arc::new(table.table.clone()),
+            table_schema: Arc::new(table_schema),
+            namespace: Arc::new(ns.namespace.clone()),
+            candidate: PartitionParam {
+                partition_id: partition.partition.id,
+                sequencer_id: partition.partition.sequencer_id,
+                namespace_id: ns.namespace.id,
+                table_id: partition.partition.table_id,
+            },
+            sort_key: Some(sort_key),                   //NGA TODO
+            partition_key: PartitionKey::from("to-do"), // NGA TODO
```

Comment:
```suggestion
            sort_key: partition.sort_key(),
            partition_key: partition.partition_key.clone(),
```
I have not checked syntax

Reply:
🤦🏻‍♀️ great catch!